### ISSUE TITLE
Persist data in local storage to limit API calls

### ISSUE THREAD

**Issue by AbhayVAshokan** [AbhayVAshokan/Hack-NASA-using-HTML#53]
Currently, an API call is made for each individual user. The same set of API calls are made each time the page is reloaded. This can be avoided by caching the results in `localStorage`. This also avoids the possibility of reaching the API limit.

**Comment by joe733** [comment]
Hi, I'm still learning JS, not there yet. So.. when implementing a search, search space should be this local storage?

**Comment by AbhayVAshokan** [comment]
Nope, search query need not be stored in localStorage. This feature is just for throttling the API. It doesn't affect the search. 

Some refactoring needs to be done to implement the search feature Append all the cards inside an array, perform a filter using the search query, and then render. A similar refactor also needs to be done in this issue. I'll try resolving this issue ASAP; then implementing the search would be easy.

**Comment by joe733** [comment]
Okay 👍🏻
